### ISSUE TITLE
Ridiculous amount of `statfs` syscalls.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#26215]
About 10 seconds:

```
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 66,48    0,371504          33     11004           statfs
 22,26    0,124408          38      3226           openat
  5,51    0,030767          37       821        81 stat
  5,28    0,029505          43       672        10 unlink
  0,16    0,000890          49        18         9 mkdir
  0,14    0,000781          41        19           rename
  0,09    0,000507          50        10           rmdir
  0,05    0,000267          29         9           utime
  0,03    0,000162          81         2           getcwd
------ ----------- ----------- --------- --------- ----------------
100.00    0,558791                 15781       100 total
```

**Comment by alexey-milovidov** [comment]
CC @alesapin

**Comment by alexey-milovidov** [comment]
Side note: Linux is capable of doing 10 million simple syscalls per second. But you should not aim this number.

**Comment by dgloeckner** [comment]
We've seen a similar behaviour in our CH server (21.6.6.51). 

Bellow you find the CH syscalls just for one BgSchPool thread. It almost uses 1 second of CPU time in an interval of 10 seconds.
Together, all 16 BgSchPool threads of a single CH server account for a CPU usage of 150%.


What about the calls to clock_gettime? 10K calls per second per thread seems quite a lot, even if the actual call is cheap.

`
BgSchPool (3883774), 378054 events, 3.9%, 1064.116 msec

   syscall            calls    total       min       avg       max      stddev
                               (msec)    (msec)    (msec)    (msec)        (%)
   --------------- -------- --------- --------- --------- ---------     ------
   futex              65182  8937.757     0.000     0.137     2.624      0.86%
   clock_gettime     100329   178.109     0.002     0.002     0.023      0.09%
   statfs             12291   127.551     0.008     0.010     0.145      0.22%
   getcpu                52     0.091     0.002     0.002     0.003      2.49%
   madvise                1     0.009     0.009     0.009     0.009      0.00%

BgSchPool/D (3883777), 2350444 events, 24.4%, 5254.705 msec

   syscall            calls    total       min       avg       max      stddev
                               (msec)    (msec)    (msec)    (msec)        (%)
   --------------- -------- --------- --------- --------- ---------     ------
   futex             386359  5552.031     0.002     0.014    46.520      1.40%
   clock_gettime     733103  1283.694     0.002     0.002     0.138      0.04%
   madvise               90     0.840     0.002     0.009     0.027      5.75%
   getcpu                 1     0.002     0.002     0.002     0.002      0.00%
`

**Comment by filimonov** [comment]
> What about the calls to clock_gettime? 

#27492

**Comment by vumdao** [comment]
Hi @alexey-milovidov 
Is this issue fixed? I'm facing this issue with clickhouse version `21.3.14.1` .

**Comment by alexey-milovidov** [comment]
It is not fixed.

According to your report, statfs only takes 3.74% of time for 1885 calls. It is not an issue.

**Comment by vumdao** [comment]
@alexey-milovidov My case is similar to [comment](https://github.com/ClickHouse/ClickHouse/issues/26215#issuecomment-878932572), it's about `syscall` to  `clock_gettime`.
16 BgSchPool threads of a single CH server account for a CPU usage of 600%.
![2021-08-30_10-49](https://user-images.githubusercontent.com/37215642/131282618-c3206c74-5392-4e9e-b2a0-7779a5f9d261.png)

```
25681   BgSchPool
25682   BgSchPool
25683   BgSchPool
25694   BgSchPool
25696   BgSchPool
25709   BgSchPool
25710   BgSchPool
25711   BgSchPool
25712   BgSchPool
25713   BgSchPool
25714   BgSchPool
25715   BgSchPool
25716   BgSchPool
25717   BgSchPool
25718   BgSchPool
25719   BgSchPool
25720   BgSchPool/D
```
Please help to advice!

**Comment by alexey-milovidov** [comment]
Syscalls to clock_gettime are removed in testing release 21.10.

**Comment by vumdao** [comment]
@alexey-milovidov I have a question that would like you favor to help me understand.

I'm using single CH with `MergeTree` engine method only so is it possible to set background_schedule_pool_size = 0 as the document says `background_schedule_pool_size` is the number of threads performing background tasks for replicated tables, Kafka streaming, DNS cache updates

Thanks

**Comment by alexey-milovidov** [comment]
@vumdao No, it is not possible. `BackgroundSchedulePool` is used everywhere in the system, including Atomic database engine and non-replicated MergeTree.

**Comment by alexey-milovidov** [comment]
Yes, you should install the latest testing release 21.10.

```
curl -O 'https://builds.clickhouse.tech/master/amd64/clickhouse' && chmod a+x clickhouse
sudo ./clickhouse install
```

**Comment by den-crane** [comment]
>Vu Dao: I have 158015 tables from 935 CH databases,
